ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/abstractrecord/RecordTypeManager.java

```diff
@@ -67,7 +67,8 @@ public synchronized int getType (Class<? extends AbstractRecord> c)
     {
         for (int i = 0; i < _map.size(); i++)
         {
-            if (_map.get(i).getRecordClass().equals(c))
+            RecordTypeMap typeMap = _map.get(i);
```

Comment:
you could avoid the second call to getRecordClass by doing
```
AbstractRecord rec = _map.get(i).getRecordClass()
```
unless you think _map.get(i) could return null in which case you need to be checking for that too.

Reply:
I am not altogether sure of what is going on here, is it possible for typeMap to be null in practice? If not perhaps we should add an exclusion for this bit. If it is possible to be null then I think we would report that elsewhere too if possible

Reply:
@mmusgrov agree, thanks!
@tomjenkinson  the issue here is that the `getRecordClass()` can return null, e.g. for https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java#L58

Reply:
@mmusgrov agree, thanks!
@tomjenkinson the issue here is that the getRecordClass() can return null, e.g. for https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java#L58